invisible_cities/core/fit_functions.py

```diff
@@ -4,16 +4,55 @@
 GML November 2016
 """
 
-import numpy  as np
-import pandas as pd
+import numpy   as np
+import pandas  as pd
+import inspect as insp
 import scipy.optimize
 import scipy.stats
 
+from functools import wraps
+
 from .                    import core_functions as coref
 from .  stat_functions    import poisson_sigma
 from .. evm.ic_containers import FitFunction
 
 
+def fixed_parameters(fn, **kwargs):
+    """
+    Wrapper to fix parameters of a function
+    which can then be used in curve_fit.
+    """
+    fn_sig    = insp.signature(fn)
+    fn_pars   = np.array(list(fn_sig.parameters))[1:]
+    par_gen   = (kwargs[k] if k in kwargs.keys() else np.nan for k in fn_pars)
+    all_args  = np.fromiter(par_gen, np.float)
+    free_pars = np.isnan(all_args)
+    
+    if np.all(free_pars):
+        raise ValueError(kwargs.keys() + " not parameters of " + fn.__name__)
+    elif np.count_nonzero(free_pars) > len(fn_pars) - len(kwargs):
+        raise ValueError("Some parameters not found in " + fn.__name__)
+    
+    @wraps(fn)
+    def fixed_fn(x, *pars):
+        all_args[free_pars] = pars
+        func_value = fn(x, *all_args)
+        ## hack to maintain spe_functions underlying decorator
+        try:
+            fixed_fn.n_gaussians = fn.n_gaussians
+        except AttributeError: pass
```

Comment:
I am not sure whether it is better
```
try:
    do_something_with_something_that_may_have_an_attribute
except AttributeError:
    pass
```
or
```
if hasattr(something_that_may_have_an_attribute, attribute):
    do_something
```

In this case I don't think it makes any difference, but it could in more complex cases, so I would go for the latter.